sqlmesh/core/loader.py

```diff
@@ -61,6 +67,23 @@ def update_model_schemas(
             schema.add_table(name, columns_to_types, dialect=model.dialect)
 
 
+def with_mocked_dependencies(loaded_model: Model) -> Model:
+    if isinstance(loaded_model, SqlModel):
+        # We mock dependencies so they can be referenced in macros before they're loaded
+        loaded_model._query_renderer._models = UniqueKeyDict(
+            "mocked_models",
+            {
+                name: _Model(  # type: ignore
+                    name=name,
+                    columns_to_types_={SQLMESH_MOCKED_COLUMN: exp.DataType.build("unknown")},
```

Comment:
Adding a single column-type pair here was necessary in order to avoid queries with no projections, since such queries are [deemed invalid](https://github.com/TobikoData/sqlmesh/blob/8b0e3cfd13cc0c6c181a2bc8e4c624ccc5a77940/sqlmesh/core/model/definition.py#L1034) and so we'd crash if we allowed them.